### PR TITLE
remove numpy from ClassificationHead to make it GPU compatible

### PR DESCRIPTION
**Pull request by dcfidalgo**
This PR should solve #145 
@dvsrepo wants to do a test before merging this one.

### CODE REVIEW

**Comment by dvsrepo** [comment]
Hi!

I have tested and got a new kind of error in the torch.cat call:
```
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
<ipython-input-15-b27213f91819> in <module>()
      4     training='/content/drive/My Drive/tweets/train-cat.yml',
      5     validation='/content/drive/My Drive/tweets/validation-cat.yml',
----> 6     verbose=True
      7 )

9 frames
/usr/local/lib/python3.6/dist-packages/biome/text/modules/heads/classification/defs.py in labels_with_probabilities(probabilities)
     78             """
     79             all_classes_probs = torch.cat(
---> 80                 [probabilities, torch.zeros(self.num_labels - len(probabilities))]
     81             )
     82             sorted_indexes_by_prob = torch.argsort(all_classes_probs, descending=True).tolist()

RuntimeError: All input tensors must be on the same device. Received cpu and cuda:0
```

**Comment by dcfidalgo** [comment]
Just fixed it. I tested it on Colab with our *examples/1.text_classifier* and it worked.
@dvsrepo Can you nevertheless run your twitter experiment to see if the results actually make sense?

**Comment by dvsrepo** [comment]
Perfect!

File: src/biome/text/modules/heads/classification/defs.py [hunk 1]
```diff
@@ -72,15 +71,21 @@ def task_name(self) -> TaskName:
...
+        def labels_with_probabilities(probabilities: torch.Tensor) -> Dict[str, float]:
             """
             Calculates the descendant sorted label + probs dictionary
             using all output classes (not only predicted)
             """
-            all_classes_probs = numpy.append(
-                probabilities, numpy.zeros(self.num_labels - len(probabilities))
+            all_classes_probs = torch.zeros(
+                self.num_labels,
+                device=probabilities.get_device()
```
[the remark this answers is not in the document]
Comment (dvsrepo) [reply src/biome/text/modules/heads/classification/defs.py:81]: @dcfidalgo have you taken a look into this suggestion?

Let me know if you make some changes so I can test and we can merge this

[the remark this answers is not in the document]
Comment (dcfidalgo) [reply src/biome/text/modules/heads/classification/defs.py:81]: @frascuchon not sure if i fully understand your comment. Maybe we can discuss it shortly today, but `.get_device()` as far as i understand just returns the id of the device on which the tensor resides.

[the remark this answers is not in the document]
Comment (dcfidalgo) [reply src/biome/text/modules/heads/classification/defs.py:81]: @frascuchon Ah, if you are referring to the if clause afterwards, it is a bit unfortunate in my opinion. `.get_device()` returns `-1` for CPU, but `torch.Tensor()` expects a `None` for `device` if it is created on a CPU ... that is the reason for the if clause.

**Pull request merged**